src/sha256.nr

```diff
@@ -191,13 +189,10 @@ unconstrained fn build_msg_block<let N: u32>(
     for i in 0..int_input {
         let mut msg_item: u32 = 0;
         // Always construct the integer as 4 bytes, even if it means going beyond the input.
+        // k is always less
         for j in 0..INT_SIZE {
             let k = i * INT_SIZE + j;
-            let msg_byte = if k < block_input {
```

Comment:
unless I am mistaken, k is always less than block_input?

Reply:
oh no it is not, I missed the int_input increment, so I restored the previous version.
We could try to peel the last iteration though.